docs/api/connection.rst

```diff
@@ -0,0 +1,278 @@
+        addresses were specified, this parameter may specify an
+        array of port numbers of the same length as the host array,
+        or it may specify a single port number to be used for all host
+        addresses.
+
+        If not specified, the value parsed from the *dsn* argument is used,
+        or the value of the ``EDGEB_PORT`` environment variable, or ``5656``
+        if neither is specified.
+
+    :param boolean options.admin:
+        If ``true``, try to connect to the special administration socket.
+
+    :param string options.user:
+        The name of the database role used for authentication.
+
+        If not specified, the value parsed from the *dsn* argument is used,
+        or the value of the ``EDGEDB_USER`` environment variable, or the
+        operating system name of the user running the application.
+
+    :param string options.database:
+        The name of the database to connect to.
+
+        If not specified, the value parsed from the *dsn* argument is used,
+        or the value of the ``EDGEDB_DATABASE`` environment variable, or the
+        operating system name of the user running the application.
+
+    :param string options.password:
+        Password to be used for authentication, if the server requires
+        one.  If not specified, the value parsed from the *dsn* argument
+        is used, or the value of the ``EDGEDB_PASSWORD`` environment variable.
+        Note that the use of the environment variable is discouraged as
+        other users and applications may be able to read it without needing
+        specific privileges.
+
+    :param number options.timeout:
+        Connection timeout in seconds.
+
+    :param callback:
+        A callback function that will be invoked when the conenction is ready.
+        The callback funciton should be of the form ``function(error,
+        connection)``. The *connection* is an instance of
+        :js:class:`Connection`.
+
+    :returns:
+        There are two ways of creating an EdgeDB connection: a Promise-based
+        approach and a callback-based. When a *callback* argument is provided,
+        the function does not return anything and instead uses the *callback*.
+        Otherwise, a ``Promise`` of an :js:class:`AwaitConnection` is returned.
+
+    Example:
+
+    .. code-block:: js
+
+        // Use the Node.js assert library to test results.
+        const assert = require("assert");
+        const edgedb = require("edgedb");
+        const connect = edgedb.default;
+        var conn;
+
+        async function main() {
+          conn = await connect({
+            dsn: "edgedb://edgedb@localhost/"
+          });
+
+          let data = await conn.fetchOne("SELECT 1 + 1");
+
+          // The result is a number 2.
+          assert(typeof data === "number");
+          assert(data === 2);
+        }
+
+        main().finally(() => {
+          // Close the connection.
+          conn.close();
```

Comment:
I think that this is also technically incorrect, as you're supposed to return Promise to chain them -- in your case you're starting an async operation (close) but not waiting for it to complete.  Again, better to replace with try..finally in the main function.  Then this snippet becomes `main()` -- short and sweet.

Reply:
I've replaced this pattern in all code samples.